### DISCUSSION TITLE
using `await asyncio.get_running_loop().run_in_executor` in post and get behaves weird

### DISCUSSION THREAD

**Discussion by axiangcoding** [fastapi/fastapi#9911] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import asyncio
import time

from fastapi import FastAPI

app = FastAPI()


@app.get("/get")
async def test_get():
    print("start test_get")
    await asyncio.get_running_loop().run_in_executor(None, do_something)
    print("end test_get")
    return {"Hello": "World"}


@app.post("/post")
async def test_post():
    print("start test_post")
    await asyncio.get_running_loop().run_in_executor(None, do_something)
    print("end test_post")
    return {"Hello": "World"}


def do_something():
    print("sleep 10s")
    time.sleep(10)
```

### Description

- call `/get` 3 times, log as follow
  ```
  start test_get
  sleep 10s
  end test_get
  INFO:     127.0.0.1:62211 - "GET /get HTTP/1.1" 200 OK
  start test_get
  sleep 10s
  end test_get
  INFO:     127.0.0.1:62245 - "GET /get HTTP/1.1" 200 OK
  start test_get
  sleep 10s
  end test_get
  INFO:     127.0.0.1:62261 - "GET /get HTTP/1.1" 200 OK
  ```
- call `/post` 3 times, log as follow
  ```
  start test_post
  sleep 10s
  start test_post
  sleep 10s
  start test_post
  sleep 10s
  end test_post
  INFO:     127.0.0.1:62323 - "POST /post HTTP/1.1" 200 OK
  end test_post
  INFO:     127.0.0.1:62325 - "POST /post HTTP/1.1" 200 OK
  end test_post
  INFO:     127.0.0.1:62328 - "POST /post HTTP/1.1" 200 OK
  ```
- Why it is executed serially in the `get` method, but it is executed in parallel in the `post` method?

### Operating System

Linux, Windows

### FastAPI Version

0.97.0

### Python Version

Python 3.11.4

### Additional Context

uvicorn==0.22.0

**Comment by IterableTrucks** [comment]
I tested with your code and called 3 /get simultaneously and it was executed in parallel.

**Reply by axiangcoding** [reply]
which is why i say this is weird, did you use the same version fastapi and python?
additional, im using uvicorn==0.22.0

**Reply by IterableTrucks** [reply]
Yes, I tested with same versions. There was nothing unexpected happened.

**Reply by axiangcoding** [reply]
But I can reproduce this weird thing every time. Can you tell me the command you use to start fastapi, I want to rule out our differences

**Comment by Kludex** [comment]
What's the event loop?

**Reply by axiangcoding** [reply]
I don't understand, i did not define the event loop somewhere else. demo is all the code.
Am I misuse the feature of fastapi?

**Reply by Kludex** [reply]
What `pip show uvloop` tells you?

**Reply by axiangcoding** [reply]
```
WARNING: Package(s) not found: uvloop
```

but i may know why this happend, my testing method is not rigorous

**Comment by axiangcoding** [comment, marked as answer]
I probably know why this happens, i test 3 calls in  same browser, i just find out this https://stackoverflow.com/questions/71516140/fastapi-runs-api-calls-in-serial-instead-of-parallel-fashion

> Important Note
When you call your endpoint for the second (third, and so on) time, please remember to do that from a tab that is isolated from the browser's main session; otherwise, succeeding requests (i.e., coming after the first one) will be blocked by the browser (on client side), as the browser will be waiting for response from the server for the previous request before sending the next one. You can confirm that by using print(request.client) inside the endpoint, where you would see the hostname and port number being the same for all incoming requests—if requests were initiated from tabs opened in the same browser window/session)—and hence, those requests would be processed sequentially, because of the browser sending them sequentially in the first place. To solve this, you could either:

when i use 3 different browser to test the `/get` call, everything works as expect
